app/cdash/public/api/v1/addBuild.php

```diff
@@ -29,11 +29,9 @@
 $project = just_get_project_from_request();
 $project->Fill();
 
-$authtoken = new AuthToken();
-if ($project->AuthenticateSubmissions && !$authtoken->validForProject($project->Id)) {
-    return;
-} elseif (!$project->AuthenticateSubmissions && !can_access_project($project->Id)) {
-    return;
+$token_hash = AuthTokenService::hashToken(AuthTokenService::getBearerToken());
```

Comment:
Do we still need these explicit token checks here (and in `relateBuilds.php`) or can we remove them and rely on the global check in `CheckPassword.php`?

This is partly a philosophical question since these two API endpoints are very "submit-like".
* `addBuild` (as the name hopefully implies) creates a new empty build.
* `relateBuilds` is used to mark connections between two builds (such as one build reusing the results of another).

Reply:
It is not entirely clear to me how `CheckPassword.php` gets executed, and whether it gets executed before this endpoint.  I just blindly swapped out the previous check with my new check, so there was no particular thought behind it.

As for the philosophical part of the question, my intuition says that submit-only tokens should be only for submissions, and nothing else, which means that these checks should be removed I think.  I'm not aware of enough of the big picture to see what the benefits/downsides of each argument are so I'll defer to your judgement here.

Reply:
Sounds good. I'm okay with deferring the removal of these previously existing blocks of code until after this PR is merged.